### PR TITLE
upgrade go version to 1.21

### PR DESCRIPTION
**Pull request by rksharma95**

### CODE REVIEW

File: .github/workflows/ci-go.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
 
       - uses: actions/setup-go@v2
         with:
-          go-version: v1.20.x
```
Comment (daemon1024) [inline .github/workflows/ci-go.yml:18]: Can we change this to take go version from go.mod? You can checkout what we did in KubeArmor PR https://github.com/kubearmor/KubeArmor/pull/1663

**Review by daemon1024 — CHANGES_REQUESTED** [review]

File: relay-server/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/kubearmor/kubearmor-relay-server/relay-server
 
-go 1.15
+go 1.21
```
Comment (daemon1024) [inline relay-server/go.mod:3]: I don't think we run snyk here so it's fine to specify go 1.22 here

File: relay-server/go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/kubearmor/kubearmor-relay-server/relay-server
 
-go 1.15
+go 1.22
```
Comment (daemon1024) [inline relay-server/go.mod:3]: Did we verify if this is enough? We might need to mention 1.22.1 in toolchain

Comment (rksharma95) [reply relay-server/go.mod:3]: it should be fine, i can build locally without toolchain specified.

**Review by daemon1024 — APPROVED** [review]

**Pull request merged**